### PR TITLE
Initialise websocket after historic data received

### LINKED ISSUE #470: Live data is collected when historic data fetch fails.
Causes errors first seen in #464 
Behaviour changed in #455. 
Revert to only initialising websocket on historic data success?

### PR DESCRIPTION
**Pull request by rjmcneill**
Closes #470

### CODE REVIEW

File: src/assets/js/data/coinbase/streaming/webSocket.js [hunk 1]
```diff
@@ -9,6 +9,7 @@ export default function() {
     var dispatch = d3.dispatch('open', 'close', 'error', 'message');
     var messageType = 'match';
     var socket;
+    var historicDataLoaded = false;
```
Comment (jleft) [inline src/assets/js/data/coinbase/streaming/webSocket.js]: I don't think the WebSocket should be concerned about whether historic data has been loaded or not - this module would have higher cohesion and be more loosely coupled if wasn't concerned with this.

**Comment by jleft** [comment]
Thanks @rjmcneill - I'd consider looking to make this change in `dataInterface`, it feels like its responsibility to me. Would something like this work?

```js
source.historicFeed(callbackGenerator(function(historicFeedError, newData) {
    if (!historicFeedError) {
        data = dateSortAscending(newData);
        dispatch[event.historicDataLoaded](data, source);

        // Could pull this out into a function...
        if (source.streamingFeed != null) {
            source.streamingFeed.on('message', function(trade) {
                _collectOhlc(data, trade);
                dispatch[event.newTrade](data, source);
            })
            .on('error', function(streamingFeedError) {
                dispatch[event.streamingFeedError](streamingFeedError, source);
            })
            .on('close', function(closeEvent) {
                dispatch[event.streamingFeedClose](closeEvent, source);
            });

            source.streamingFeed();
        }
    } else {
        dispatch[event.historicFeedError](historicFeedError, source);
    }
}));
```

**Comment by rjmcneill** [comment]
@jleft yup, that looks much easier!
I agree with you that this is a better approach as well, keeping them loosely coupled

**Comment by jleft** [comment]
:+1: LGTM

**Pull request merged**